src/test/fuzz/crypto_diff_fuzz_chacha20.cpp

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) 2020-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <crypto/chacha20.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+
+#include <cstdint>
+#include <vector>
+
+/*
+From https://cr.yp.to/chacha.html
+chacha-merged.c version 20080118
+D. J. Bernstein
+Public domain.
+*/
+
+typedef unsigned int u32;
+typedef unsigned char u8;
+
+#define U8C(v) (v##U)
+#define U32C(v) (v##U)
+
+#define U8V(v) ((u8)(v)&U8C(0xFF))
+#define U32V(v) ((u32)(v)&U32C(0xFFFFFFFF))
+
+#define ROTL32(v, n) (U32V((v) << (n)) | ((v) >> (32 - (n))))
+
+#define U8TO32_LITTLE(p)                                              \
+    (((u32)((p)[0])) | ((u32)((p)[1]) << 8) | ((u32)((p)[2]) << 16) | \
+     ((u32)((p)[3]) << 24))
+
+#define U32TO8_LITTLE(p, v)      \
+    do {                         \
+        (p)[0] = U8V((v));       \
+        (p)[1] = U8V((v) >> 8);  \
+        (p)[2] = U8V((v) >> 16); \
+        (p)[3] = U8V((v) >> 24); \
+    } while (0)
+
+/* ------------------------------------------------------------------------- */
+/* Data structures */
+
+typedef struct
+{
+    u32 input[16];
+} ECRYPT_ctx;
+
+/* ------------------------------------------------------------------------- */
+/* Mandatory functions */
+
+void ECRYPT_keysetup(
+    ECRYPT_ctx* ctx,
+    const u8* key,
+    u32 keysize, /* Key size in bits. */
+    u32 ivsize); /* IV size in bits. */
+
+void ECRYPT_ivsetup(
+    ECRYPT_ctx* ctx,
+    const u8* iv);
+
+void ECRYPT_encrypt_bytes(
+    ECRYPT_ctx* ctx,
+    const u8* plaintext,
+    u8* ciphertext,
+    u32 msglen); /* Message length in bytes. */
+
+/* ------------------------------------------------------------------------- */
+
+/* Optional features */
+
+void ECRYPT_keystream_bytes(
+    ECRYPT_ctx* ctx,
+    u8* keystream,
+    u32 length); /* Length of keystream in bytes. */
+
+/* ------------------------------------------------------------------------- */
+
+#define ROTATE(v, c) (ROTL32(v, c))
+#define XOR(v, w) ((v) ^ (w))
+#define PLUS(v, w) (U32V((v) + (w)))
+#define PLUSONE(v) (PLUS((v), 1))
+
+#define QUARTERROUND(a, b, c, d) \
+    a = PLUS(a, b);              \
+    d = ROTATE(XOR(d, a), 16);   \
+    c = PLUS(c, d);              \
+    b = ROTATE(XOR(b, c), 12);   \
+    a = PLUS(a, b);              \
+    d = ROTATE(XOR(d, a), 8);    \
+    c = PLUS(c, d);              \
+    b = ROTATE(XOR(b, c), 7);
```

Comment:
```suggestion
    a = PLUS(a, b); d = ROTATE(XOR(d, a), 16);   \
    c = PLUS(c, d); b = ROTATE(XOR(b, c), 12);   \
    a = PLUS(a, b); d = ROTATE(XOR(d, a), 8);    \
    c = PLUS(c, d); b = ROTATE(XOR(b, c), 7);
```
nit: this pattern of defining the `QUATERROUND` macro is following in many places ([chacha20.cpp](https://github.com/bitcoin/bitcoin/blob/master/src/crypto/chacha20.cpp#L15)).

Reply:
Thanks! I've made the changes.